tests/job/test_genericJob.py

```diff
@@ -502,5 +503,25 @@ def test_return_codes(self):
             pass
         self.assertTrue(j.status.aborted, "Job did not abort even though return code is 2!")
 
+    def test_job_executor(self):
+        j = self.project.create_job(ReturnCodeJob, "job_with_executor")
+        j.input["accepted_codes"] = [1]
+        j.server.executor = ProcessPoolExecutor()
+        self.assertTrue(j.server.run_mode.executor)
+        fs = j.run()
+        fs.result()
+        self.assertTrue(fs.done())
+
+    def test_job_executor_copy(self):
+        j1 = self.project.create_job(ReturnCodeJob, "job_with_executor_copy")
+        j1.input["accepted_codes"] = [1]
+        j1.server.executor = ProcessPoolExecutor()
+        j2 = j1.copy()
+        self.assertTrue(j2.server.run_mode.executor)
```

Comment:
```suggestion
        self.assertIs(j2.server.executor, j.server.executor)
        self.assertTrue(j2.server.run_mode.executor)
```

I believe we also want them to literally have the same executor, so let's check that too.